contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_v4l2_video_decoder_performance.py

```diff
@@ -0,0 +1,237 @@
+        logging.info(ret.stdout)
+        return ret.stdout
+    except Exception as e:
+        logging.error(e.stderr)
+        raise SystemExit(e.returncode)
+
+
+def is_valid_result(input_text: str, min_fps: float) -> bool:
+    """
+    Extracts the last-message value from the given input string.
+    Example
+            last-message = rendered: 98, dropped: 0, current: 95.53,
+                average: 98.43
+    Pass criteria
+        1. The value of dropped frame must be 0
+        2. The value of average fps must greater than or equal to min_fps
+
+    :param input_text:
+        The input string containing the data of last-message.
+
+    :param min_fps:
+        A value that all average FPS must not fall below
+
+    :returns:
+        True if the result meets the pass criteria; false otherwise .
+    """
+    # Find all matches in the input text
+    pattern = re.compile(r"dropped: (\d+), current: [\d.]+, average: ([\d.]+)")
+    matches = pattern.findall(input_text)
+    if not matches:
+        logging.error("Unable to find any matching data.")
+        return False
+    for dropped, average in matches:
+        # Leave once a value doesn't match the pass criteria
+        if int(dropped) != 0 or float(average) < float(min_fps):
+            logging.error("Found values that violate the pass criteria.")
+            return False
+    return True
+
+
+def main() -> None:
+    """
+    This function performs the following steps:
+
+    1. Checks if the golden sample file exist.
+    2. Builds a GStreamer command to process the golden sample using the
+        specified decoder.
+    3. Executes the command and get the outcome back
+    4. Judge the outcome to see if it meets the Pass Criteria
+
+    :param args:
+        An object containing the following attributes:
+            - `golden_sample_path` (str): The path to the golden sample file.
+            - `decoder_plugin` (str): The video decoder to use, e.g.,
+                "v4l2vp8dec", "v4l2vp9dec".
+            - `minimum_average_fps` (str): The minimum value of FPS
+                that all average FPS value should not violate
+
+    :raises SystemExit:
+        If the golden sample file does not exist, or if the outcome violates
+        the pass criteria.
+    """
+    args = register_arguments()
+    logging.info(
+        "Pass Criteria"
+        "\n 1. All dropped frames must be 0"
+        "\n 2. All average fps values must greater than or equal to {}".format(
+            args.minimum_fps
+        )
+    )
+    # Check the golden sample exixt
+    if not os.path.exists(args.golden_sample_path):
+        raise SystemExit(
+            "Golden Sample '{}' doesn't exist".format(args.golden_sample_path)
+        )
+    gst_launch_bin = os.getenv("GST_LAUNCH_BIN", "gst-launch-1.0")
+    cmd = build_gst_command(
+        gst_bin=gst_launch_bin,
+        golden_sample_path=args.golden_sample_path,
+        decoder=args.decoder_plugin,
+        sink=args.sink,
+    )
+
```

Comment:
```
src_file, function = query_func(args.performance_mode_targret)
test_func = importlib(src_file, funcion)
with test_func():
  output = execute_command(cmd).rstrip(os.linesep)
```

Reply:
I updated the way to import external python script. Let's discuss it when you back.